coalib/results/result_actions/IgnoreResultAction.py

```diff
@@ -57,11 +57,31 @@ def apply(self, result, original_file_dict, file_diff_dict, language: str,
         filename = source_range.file
 
         ignore_diff = Diff(original_file_dict[filename])
-        ignore_diff.change_line(
-            source_range.start.line,
-            original_file_dict[filename][source_range.start.line-1],
-            original_file_dict[filename][source_range.start.line-1].rstrip() +
-            '  ' + ignore_comment)
+        """
+        Apply the ignore comment using 'and' with the previous ignore,
+        if the same line already has a previous ignore comment.
+        """
+
+        if 'Ignore' in str(original_file_dict[filename][
+                    source_range.start.line-1].rstrip()) and 'Bear' in str(
+                    original_file_dict[filename][
+                        source_range.start.line-1].rstrip()):
+            ignore_comment = ignore_comment[
+                ignore_comment.index('Ignore'):].replace('Ignore', ' and')
+            initial_line = original_file_dict[filename][
+                source_range.start.line-1].rstrip() + ' '
+            new_line = initial_line[:-(len(
+                initial_line) - initial_line.rfind('Bear') - len('Bear'))]
+            ignore_diff.change_line(
+                source_range.start.line,
+                original_file_dict[filename][source_range.start.line-1],
+                new_line + ignore_comment)
+        else:
+            ignore_diff.change_line(
+                source_range.start.line,
+                original_file_dict[filename][source_range.start.line-1],
+                original_file_dict[filename][
+                    source_range.start.line-1].rstrip() + '  ' + ignore_comment)
```

Comment:
In any way you are going to change a line here, right?
I would move the `change_line` part out of the if, and inside the if you only compute the patched line / ignore comment.

Reply:
So you would do (roughly):
```
if ignore_comment_existing:
    # assemble more complex ignore comment using 'and'
else:
    # assemble fist ignore comment

ignore_diff.change_line(..., line_with_ignore_comment)
```